hs_core/hydroshare/resource.py

```diff
@@ -354,11 +359,13 @@ def create_resource(
 
         fed_zone_home_path = ''
         if fed_res_path:
-            fed_zone_home_path = fed_res_path
             resource.resource_federation_path = fed_res_path
+            fed_zone_home_path = fed_res_path
             resource.save()
-        elif fed_res_file_names:
-            fed_zone_home_path = utils.get_federated_zone_home_path(fed_res_file_names[0])
+
+        # TODO: this option should be eliminated. Require explicit zone path
+        elif len(source_names) > 0:
```

Comment:
@alvacouch I understand your point, but at this point, iRODS browser only allow users to browse into the same zone, if users want to browse into another zone, they have to login again, which clears up the previous iRODS file selection and restarts another iRODS file selection. In short, we don't allow users to browse into multiple different zones at the same time as is implemented currently. So you can update your TO DO comments to reflect this.